src/reducers/dialogVisibility.js

```diff
@@ -0,0 +1,25 @@
+import Immutable from 'immutable'
+import { actionReducer } from './utils'
+
+const dialogVisibility = actionReducer(Immutable.fromJS({ showVmDialog: false, showVmDetail: false, showEditTemplate: false }), {
```

Comment:
Now considering how this will scale for the future when we will have even more dialog types I believe a bit different store representation would be more suitable. 

One alternative would be to have a simple `dialog` field there with a string identifying the type of dialog that is to be shown (since I assume this sort of "top level" dialog will be possible to display only one at a time) with values like (`null`, `"VM_EDIT"`, `"TEMPLATE_EDIT"`, ...)

Other to have this split by 2 levels: one field identifying the general dialog and second the current "behavior" (sorry for the pun with legacy gwt UICommon model :-)) or it. But in any case I would prefer to avoid this explosion of boolean flags.....

What do you guys think: @jelkosz @mareklibra @vojtechszocs ?